File: queue-frontend/src/app/dtos/department/statistics/ResponseDepartmentDashBoardDto.ts
import { ResponseCountServicesByDepartmentsStatisticsDto } from "./ResponseCountServicesByDepartmentsStatisticsDto"
import { ResponseCountTotalDepartmentsStatisticsDto } from "./ResponseCountTotalDepartmentsStatisticsDto"
import { ResponseDepartmentPercentagesStatisticsDto } from "./ResponseDepartmentPercentagesStatisticsDto"
import { ResponseDepartmentsCreatedByMonthStatisticsDto } from "./ResponseDepartmentsCreatedByMonthStatisticsDto";

export interface ResponseDepartmentDashBoardDto {

  countTotalDepartmentsStatistics: ResponseCountTotalDepartmentsStatisticsDto;
  countServicesByDepartments: ResponseCountServicesByDepartmentsStatisticsDto[];
  departmentPercentagesStatistics: ResponseDepartmentPercentagesStatisticsDto;
  departmentsCreatedByMonthStatistics: ResponseDepartmentsCreatedByMonthStatisticsDto[]
}

File: queue/src/main/java/br/com/queue/dtos/department/statistics/ResponseDepartmentDashBoardDto.java
package br.com.queue.dtos.department.statistics;

import java.util.List;

public record ResponseDepartmentDashBoardDto(
        ResponseCountTotalDepartmentsStatisticsDto countTotalDepartmentsStatistics,
        List<ResponseCountServicesByDepartmentsStatisticsDto> countServicesByDepartments,
        ResponseDepartmentPercentagesStatisticsDto departmentPercentagesStatistics,
        List<ResponseDepartmentsCreatedByMonthStatisticsDto> departmentsCreatedByMonthStatistics
) {
}
